RFC Proposal Path (#462)
pyrsia/pyrsia: docs/rfc/0000-template.md

- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
- Start Date: (fill me in with today's date, YYYY-MM-DD)
- RFC PR: [pyrsia/pyrsia#0000](https://github.com/pyrsia/pyrsia/pull/0000)

> - Pyrsia Issue: [pyrsia/pyrsia#0000](https://github.com/pyrsia/pyrsia/issues/0000)

Reviewer: is 0000 the standard way to do it?

Author: It's just a placeholder for any number, is that unclear?

- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
- Start Date: (fill me in with today's date, YYYY-MM-DD)
- RFC PR: [pyrsia/pyrsia#0000](https://github.com/pyrsia/pyrsia/pull/0000)

> - Rust Issue: [pyrsia/pyrsia#0000](https://github.com/pyrsia/pyrsia/issues/0000)

Reviewer 2: why does it say **Rust** issue here?

Author: I absolutely stole most of this from the Rust community 

great catch

# Summary

One paragraph explanation of the feature or change.

# Motivation

Why are we doing this? What use cases does it support? What is the expected outcome?

# Guide-level explanation

Explain the proposal as if it was already included in the language and you were teaching it to another Pyrsia member. That generally means:

- Introducing new named concepts.
- Explaining the feature largely in terms of examples.
- Explaining how Pysia users should *think* about the feature, and how it should impact the way they use Pyrsia. It should explain the impact as concretely as possible.
- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
- If applicable, describe the differences between teaching this to existing Pyrsia users and new Pyrsia users.

> For implementation-oriented RFCs (e.g. for compiler internals), this section should focus on how compiler contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.

Reviewer 3: Not a good example since pyrsia does not have compiler internals. Should be removed or rephrased.

Author: ```suggestion
For implementation-oriented RFCs (e.g. for peer-to-peer networking technology stack), this section should focus on how compiler contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
```

# Reference-level explanation

This is the technical portion of the RFC. Explain the design in sufficient detail that:

- Its interaction with other features is clear.
- It is reasonably clear how the feature would be implemented.
- Corner cases are dissected by example.

The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.

# Drawbacks

> Why should we *not* do this?

Reviewer 3: Instead, 
what will be the loss if we do this?

Why should we not do this? does not read the same as drawbacks to me.

Author: > Why should we not do this? does not read the same as drawbacks to me.

🤔 Thats exactly how I learnt "drawback"

> plural noun: drawbacks
> 1. a feature that renders something less acceptable; a disadvantage or problem.
>    "the main drawback of fitting catalytic converters is the cost"

*** 

@betarelease  How do you feel about 

```suggestion
Why should we *not* do this? What are the disadvantages of the propose? What negative side effects or limitation would be consequence?
```

Is that any better?

Author: Yes!

Author: ```suggestion
Why should we *not* do this? What are the trade-offs? What negative side effects or limitation could be a consequence?
```

# Rationale and alternatives

- Why is this design the best in the space of possible designs?
- What other designs have been considered and what is the rationale for not choosing them?
- What is the impact of not doing this?

# Prior art

Discuss prior art, both the good and the bad, in relation to this proposal.
A few examples of what this can include are:

- Does this feature exist in other package managers and what experience have their community had?
- For community proposals: Is this done by some other community and what were their experiences with it?
- For other teams: What lessons can we learn from what other communities have done here?

> - Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.

Reviewer 3: Rephrase to `references` (In Rust language context it might be papers but for Pyrsia it is papers but more importantly real web implementation references)

Author: I agree, this can be improved.

With the blockchain specifically there are tons of papers. So this made sense, but in the wider picture there needs to be more language 

👍

Author: ```suggestion
- References: Are there any published papers or great posts that discuss this? If you have some relevant research to refer to, this can serve as a more detailed theoretical background. Existing implementation references for practical background can also help fill in a proposal.
```

This is an excellent place to share research notes, articales or proof of concepts that can support or educate the reviewers.

# Unresolved questions

- What parts of the design do you expect to resolve through the RFC process before this gets merged?
- What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
- What related issues do you consider out of scope for this RFC that could be addressed in the future independently of the solution that comes out of this RFC?

# Future possibilities

> Think about what the natural extension and evolution of your proposal would

Reviewer 3: This paragraph is too much to ask from a proposer.

Author: I made this sectional optional but I like the idea to leave space for anything extra the author might want to add

Is that a good middle ground?

be and how it would affect the project as a whole in a holistic
way. Try to use this section as a tool to more fully consider all possible
interactions with the project and language in your proposal.
Also consider how this all fits into the roadmap for the project
and of the relevant sub-team.

This is also a good place to "dump ideas", if they are out of scope for the
RFC you are writing but otherwise related.

> If you have tried and cannot think of any future possibilities,

Reviewer 3: Sounds non-inclusive(meaning condescending this case)

you may simply state that you cannot think of anything.

> Note that having something written down in the future-possibilities section

Reviewer 3: Too much structure enforced by adding this in the footnote.
